[TIMOB-19435] Change Pref Activity to Pref Frag

ashcoding opened the pull request:
Jira: https://jira.appcelerator.org/browse/TIMOB-19435

Commit:
[TIMOB-19435] Change Pref Activity to Pref Frag

hieupham007 commented on android/modules/ui/src/java/ti/modules/titanium/ui/android/TiPreferencesFragment.java:
```diff
@@ -0,0 +1,42 @@
+        super.onCreate(savedInstanceState);
+        
+        String prefsName = DEFAULT_PREFS_RNAME;
+        if (getActivity().getIntent().hasExtra("prefsName")) {
```
what if getActivity() or getIntent() is null?

hieupham007 commented on android/modules/ui/src/java/ti/modules/titanium/ui/android/TiPreferencesFragment.java:
```diff
@@ -0,0 +1,42 @@
+        
+        String prefsName = DEFAULT_PREFS_RNAME;
+        if (getActivity().getIntent().hasExtra("prefsName")) {
+            String name = getActivity().getIntent().getExtras().getString("prefsName");
```
please try to avoid duplicate method calling: getActivity().getIntent() is called twice back to back

Commit:
[TIMOB-19435] Addressed comments for Pref Frag

ashcoding commented:
Addressed the comments. I'm putting the logic to get the pref name in the activity and passing it to the fragment as an argument via a bundle.

I think this might be better.

hieupham007 commented:
This is what I see: 
![screenshot_2015-09-16-14-46-05](https://cloud.githubusercontent.com/assets/1141803/9919503/4418e6ec-5c82-11e5-83ed-57baada3540b.png)

Is this expected?

ashcoding commented:
That is not the expected outcome. May I know what device you are using and the android version?

hieupham007 commented:
Galaxy Nexus 4.1.2 and Android M Preview 3. Same results.

Commit:
Merge branch 'master' of https://github.com/appcelerator/titanium_mobile into TIMOB-19435

ashcoding commented:
Not sure why but it's working correctly for me. I pulled and merge master to make this updated with master. Could you try it again?

hieupham007 commented:
I'm seeing this now: 
![screenshot_2015-09-17-13-42-09](https://cloud.githubusercontent.com/assets/1141803/9945703/a7d9dc1a-5d43-11e5-8345-0e62f299ec89.png)

ashcoding commented:
I'm assuming the title `215` is not intended? This seems quite strange. I'll look into it more.

Commit:
[TIMOB-19435] Allow for setting of the Title

ashcoding commented:
Updated the PR. The title is now taken from the first Preference Screen title. Eg, this will be `TITLE` in the following example.
```xml
<?xml version="1.0" encoding="utf-8"?>
<PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android" android:title="TITLE">
    <CheckBoxPreference
        android:key="key1"
        android:title="this is key 1"
        android:summary="key 1 summary"
        android:defaultValue="false" />
    <CheckBoxPreference
        android:key="key2"
        android:title="this is key 2"
        android:summary="key 2 summary"
        android:defaultValue="false" />
</PreferenceScreen>
```

hieupham007 commented:
Code reviewed and functionally tested. Request **accepted**
